From: Muhammad Usama Anjum <usama.anjum@collabora.com>
Subject: selftests/mm: hugetlb-madvise: fix type mismatch issues
Date: Thu, 9 Jan 2025 22:38:35 +0500

Fix type mismatch warnings:
  hugetlb-madvise.c:30:25: warning: comparison of integer expressions of different signedness: `int' and `long unsigned int' [-Wsign-compare]
     30 |                 if (fhp != (exp_free)) {                                \
        |                         ^~
  hugetlb-madvise.c:114:9: note: in expansion of macro `validate_free_pages'
    114 |         validate_free_pages(free_hugepages - NR_HUGE_PAGES);
        |         ^~~~~~~~~~~~~~~~~~~

Link: https://lkml.kernel.org/r/20250109173842.1142376-10-usama.anjum@collabora.com
Signed-off-by: Muhammad Usama Anjum <usama.anjum@collabora.com>
Cc: Andy Lutomirski <luto@amacapital.net>
Cc: Jérôme Glisse <jglisse@redhat.com>
Cc: Kees Cook <kees@kernel.org>
Cc: Shuah Khan <shuah@kernel.org>
Cc: Will Drewry <wad@chromium.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/hugetlb-madvise.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/testing/selftests/mm/hugetlb-madvise.c~selftests-mm-hugetlb-madvise-fix-type-mismatch-issues
+++ a/tools/testing/selftests/mm/hugetlb-madvise.c
@@ -26,7 +26,7 @@
 
 #define validate_free_pages(exp_free)					\
 	do {								\
-		int fhp = get_free_hugepages();				\
+		unsigned int fhp = get_free_hugepages();		\
 		if (fhp != (exp_free)) {				\
 			printf("Unexpected number of free huge "	\
 				"pages line %d\n", __LINE__);		\
_
